apps/protoc-gen-openapi/examples/tests/mapfields/openapi.yaml

```diff
@@ -36,7 +36,30 @@ components:
             properties:
                 message_id:
                     type: string
-                labels:
+                string_list:
+                    type: array
+                    items:
+                        type: string
+                sub_message_list:
+                    type: array
+                    items:
+                        $ref: '#/components/schemas/SubMessage'
+                object_list:
+                    type: array
+                    items:
+                        type: object
+                strings_map:
```

Comment:
Should this (and the two maps below) be an object?

Reply:
Good catch!
Sorry... I went the wrong way around while coding this.
My thinking was "how to express map protos in OpenAPI?" - instead of: is this even doable in OpenAPI and how would it look.

I tried to get the code to parse sub messages and then I changed the example yaml files, once it was working...
Clearly that was not a good idea.
 
I think object_list and the other lists is correct.
`repeated google.protobuf.Struct object_list = 4;` should be:
```yaml
object_list:
    type: array
    items:
        type: object
```
It's an array (repeated) of objects (Struct) with no restrictions, right?

I think that the correct version of the maps is something like this:
```yaml
  stringsMap:
      type: object
      additionalProperties:
        type: string
  subMessagesMap:
      type: object
      additionalProperties:
        $ref: '#/components/schemas/SubMessage'
  objectsMap:
      type: object
      additionalProperties:
        type: object
```
The `object` type, naturally forces key to be a string.
The additionalProperties restricts the value of each property.
This seems to produce the correct example values in Swagger editor:
```json
{
  "messageId": "string",
  "stringList": [
    "string"
  ],
  "subMessageList": [
    {
      "id": 0,
      "label": "string"
    }
  ],
  "objectList": [
    {}
  ],
  "stringsMap": {
    "additionalProp1": "string",
    "additionalProp2": "string",
    "additionalProp3": "string"
  },
  "subMessagesMap": {
    "additionalProp1": {
      "id": 0,
      "label": "string"
    },
    "additionalProp2": {
      "id": 0,
      "label": "string"
    },
    "additionalProp3": {
      "id": 0,
      "label": "string"
    }
  },
  "objectsMap": {
    "additionalProp1": {},
    "additionalProp2": {},
    "additionalProp3": {}
  }
}
```

Maybe I should try to get map<int32,string> and the likes to be converted to arrays - if Envoy can handle that.

I will make the changes and push again.

Reply:
Thanks, I agree that using additionalProperties is a good way to constrain the type of map elements (I'd forgotten about that).